CONTRIBUTING.md

```diff
@@ -0,0 +1,291 @@
+
+Still here? Great! It can definitely happen that you have a suggestion that just doesn't fit in any style guide.
+Be very critical and perhaps do an experiment in the code first to see if your suggestion is worth having.
+If a feature can help in the projects adoption and growth we are still happy to discuss it.
+
+# Getting started with Fantomas
+
+Fantomas has fairly straightforward setup.
+
+> git clone https://github.com/yourfork/fantomas
+
+After cloning the repository you can restore the local .NET tools
+
+> dotnet tool restore
+
+Next you can execute FAKE build targets.
+
+> dotnet fake run build.fsx --list
+
+The default target will execute the CI build.
+
+# Pull request ground rules
+
+Code changes in pull requests will have a certain set of expectations.
+In general, changes should be made as consistent to the current code base as possible.
+Don't introduce unnecessary new concepts and try and change as little as code possible to achieve your goal.
+
+Always start with the mindset that you are going to introduce a change that might have an impact on how the tool behaves.
+Capture this change first in a unit test. Set your expectations in the assert part of the test before touching anymore.
+This project is very well suited for [Test-driven development](https://en.wikipedia.org/wiki/Test-driven_development) and that should be the goal.
+
+Typical unit test template:
+
+```fsharp
+[<Test>]
+let ``my new test`` () =
+    formatSourceString false """
+let myInput =     42
+"""  config
+    |> prepend newline
+    |> should equal """
+let myInput = 42
+"""
+```
+
+The vast majority of the tests uses the template listed above. Only deviate from this when necessary.
+Try and find a suiting file in `Fantomas.Tests` or introduce a new file.
+
+A new test file should look like:
+
+```fsharp
+module Fantomas.Tests.MyNewConceptTests
+
+open NUnit.Framework
+open FsUnit
+open Fantomas.Tests.TestHelper
+
+// add tests here...
+```
+
+Filename: `MyNewConceptTests.fs`.
+
+When developing a new feature, add new tests to cover all code paths.
+
+## Guidelines
+
+- Unit tests should start with a `lowercase` letter.
+- Verify if the change you are making also should apply to signature files (*.fsi).
+- Check if you need additional tests to cope with a different combination of settings.
+- Check if you need additional tests to cope with a different combination of defines (`#if DEBUG`, ...).
+- When creating a test that is linked to a GitHub issue, add the number at the back with a comma f.ex.
+- Write/update documentation when necessary.
+
+```fsharp
+[<Test>]
+let ``preserve compile directive between piped functions, 512`` () = ...
+```
+
+### Small steps
+
+Please ask yourself before contributing if you received a clear signal to proceed.
+This is too avoid frustrations with large pull requests that were never discussed.
```

Comment:
s/too/to

Reply:
I'd probably hold off on stylistic reviews for half an hour or so. I've nearly finished my own pass through it.